app/uk/gov/hmrc/alcoholdutyreturns/service/ReturnsService.scala

```diff
@@ -70,7 +73,59 @@ class ReturnsService @Inject() (
                                       returnToSubmit,
                                       returnId.appaId
                                     )
+      _                           = cacheRepository.get(returnId).map {
```

Comment:
There can be a race condition here where the clearUserAnswersById can be executed before get.

You can use something like:

    for {
      maybeTotalDutyDueByTaxType <- calculateTotalDutyDueByTaxType(returnSubmission)
      returnToSubmit              = returnConvertedToSubmissionFormat.copy(totalDutyDuebyTaxType = maybeTotalDutyDueByTaxType)
      _                          <- validateAgainstSchema(returnToSubmit)
      returnCreatedDetails       <- returnsConnector.submitReturn(
                                      returnToSubmit,
                                      returnId.appaId
                                    )
      _                          <- EitherT.right[ErrorResponse](
                                      cacheRepository
                                        .get(returnId)
                                        .map {
                                          case Some(ua) =>
                                            auditReturnSubmitted(
                                              Some(ua),
                                              returnToSubmit,
                                              AdrReturnCreatedDetails.fromReturnCreatedDetails(returnCreatedDetails),
                                              returnId
                                            )
                                          case None     =>
                                            logger.warn("User answers couldn't be retrieved while auditing return submission")
                                            auditReturnSubmitted(
                                              None,
                                              returnToSubmit,
                                              AdrReturnCreatedDetails.fromReturnCreatedDetails(returnCreatedDetails),
                                              returnId
                                            )
                                        }
                                        .recoverWith { case e: Throwable =>
                                          Future.successful(logger.warn("Encountered error while auditing return submission", e))
                                        }
                                        .andThen { case _ =>
                                          cacheRepository.clearUserAnswersById(returnId)
                                        }
                                    )
    } yield returnCreatedDetails